public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -909,6 +901,20 @@ export class PrometheusDatasource extends DataSourceWithBackend<PromQuery, PromO
     return getOriginalMetricName(labelData);
   }
 
+  enhanceExprWithAdHocFilters(expr: string) {
```

Comment:
nitpick: i think this function could be simplified to be like this:
```javascript
  enhanceExprWithAdHocFilters(expr: string) {
    const adhocFilters = this.templateSrv.getAdhocFilters(this.name);
    return adhocFilters.reduce(...
     ...
     ...
    }, expr);
  }
```

Reply:
👍 I merged this, we can always clean this up if we want to.